packages/account/src/Sections/Security/TwoFactorAuthentication/__test__/two-factor-disabled.spec.tsx

```diff
@@ -0,0 +1,129 @@
+        secret_key: 'hello123',
+        qr_secret_key: '3535',
+        is_qr_loading: false,
+    };
+
+    const store = mockStore({
+        client: {
+            has_enabled_two_fa: false,
+        },
+        ui: {
+            is_mobile: true,
+        },
+    });
+
+    const renderComponent = ({ store_config = store, mock = mock_props }) => {
+        render(
+            <StoreProvider store={store_config}>
+                <TwoFactorDisabled {...mock} />
+            </StoreProvider>
+        );
+    };
+
+    it('should render TwoFactorDisabled component if has_enabled_two_fa is false', () => {
+        renderComponent({ store_config: store });
+
+        const setup_title = screen.getByText(/How to set up 2FA for your Deriv account/i);
+        expect(setup_title).toBeInTheDocument();
+    });
+
+    it('should render timeline_1 component title ', () => {
+        renderComponent({ store_config: store });
+
+        const timeline_title_1 = screen.getByText(/Scan the QR code below with your 2FA app. We recommend./i);
+        const authy_link = screen.getByRole('link', { name: 'Authy' });
+        const google_authenticator_link = screen.getByRole('link', { name: 'Google Authenticator' });
+
+        expect(timeline_title_1).toBeInTheDocument();
+        expect(authy_link).toHaveAttribute('href', 'https://authy.com/');
+        expect(google_authenticator_link).toHaveAttribute(
+            'href',
+            'https://github.com/google/google-authenticator/wiki#implementations'
+        );
+    });
+
+    it('should render QR code', () => {
+        renderComponent({ store_config: store });
+
+        const qr_code = screen.getByText('QRCode');
+        expect(qr_code).toBeInTheDocument();
+    });
+
+    it('should render clipboard component to setup 2FA', () => {
+        renderComponent({ store_config: store });
+
+        const helper_text = screen.getByText(
+            /If you are unable to scan the QR code, you can manually enter this code instead:/i
+        );
+        const secret_text = screen.getByText('hello123');
+        const clipboard_component = screen.getByTestId('dt_2fa_clipboard');
+
+        expect(helper_text).toBeInTheDocument();
+        expect(secret_text).toBeInTheDocument();
+        expect(clipboard_component).toBeInTheDocument();
+    });
+
+    it('should render step-2 title for setting up 2FA', () => {
+        renderComponent({ store_config: store });
+
+        const step_2_title = screen.getByText(/Enter the authentication code generated by your 2FA app:/i);
+        expect(step_2_title).toBeInTheDocument();
+    });
+
+    it('should render digitform component if 2FA is disabled', () => {
+        renderComponent({ store_config: store });
+
+        const digitform = screen.getByTestId('dt_digitform_2fa_disabled');
+        expect(digitform).toBeInTheDocument();
+    });
+
+    it('should render 2FA article component for mobile', () => {
+        const new_store = {
+            ...store,
```

Comment:
I don’t really understand, why everywhere we create new store and extract there previous one.
why just not to change the fields directly in already crated mockStore?
store.client.is_mobile = true.
if for every test case we need new store, we can do it in beforeEach function.
you can ignore this :)

Reply:
We can do this as well..its just 2 diff ways of doing same thing I guess :)